### PR TITLE
update to latest mach-core API

### PR DESCRIPTION
**Pull request by alichraghi**
- [x] By selecting this checkbox, I agree to license my contributions to this project under the license(s) described in the LICENSE file, and I have the right to do so or have received permission to do so by an employer or client I am producing work for whom has this right.

### CODE REVIEW

File: src/editor/app.zig [hunk 1]
```diff
@@ -1,8 +1,10 @@
 const std = @import("std");
 const mach = @import("mach");
+const core = mach.core;
 const gpu = mach.gpu;
 
-pub const App = @This();
+pub const name = .editor;
+pub const App = mach.App(.{ mach.Module, @This() });
```
Comment (slimsag) [inline src/editor/app.zig:7]: Passing `@This()` into `mach.App` as a module is.. sketchy and seems very wrong. Why is this necessary?

Comment (slimsag) [reply src/editor/app.zig:7]: I get what you're doing - fine for now, but let's clean this up later.

**Review by slimsag — APPROVED** [review]

File: src/editor/app.zig (hunk 1 above)
Comment (alichraghi) [reply src/editor/app.zig:7]: otherwise, init, deinit and update can't be called
